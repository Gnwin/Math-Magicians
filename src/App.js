import React from 'react';
import './App.css';
import Calculator from './components/Calculator';

class App extends React.Component() {
  state = {}

  render() {
    return (
      <div className="calculator">
        <Calculator />
      </div>
    );
  }
}

export default App;
