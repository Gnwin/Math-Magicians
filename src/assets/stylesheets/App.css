.nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 2rem 0;
  margin-bottom: 3rem;
}

.nav-title {
  font-size: 3rem;
}

.nav-link {
  font-size: 1.5rem;
  padding: 0 0.8rem;
  border-left: 2px solid black;
}

.nav-link:first-child {
  padding-left: 0;
  border-left: none;
}

.nav-link:last-child {
  padding-right: 0;
}

.home-title {
  font-size: 1.7rem;
}

.calcpage {
  display: flex;
  justify-content: space-between;
}

.calctitle {
  font-size: 1.7rem;
}

.calculator {
  display: grid;
  grid-template-columns: repeat(4, auto);
  width: 55%;
}

.answer {
  width: 100%;
  grid-column-start: 1;
  grid-column-end: 5;
  text-align: right;
  padding: 0.7rem 0.6rem 0.7rem 0;
  background-color: rgb(143, 141, 141);
  color: white;
  border: 1px solid white;
}

.zero {
  grid-column-start: 1;
  grid-column-end: 3;
}

.center {
  text-align: center;
  padding: 0.7rem 2.2rem;
  background-color: rgb(227, 227, 227);
  border: 1px solid white;
  cursor: pointer;
}

.orange {
  background-color: rgb(248, 182, 59);
}

.quote {
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 55vh;
  padding: 2rem;
}

.quote-msg {
  font-size: 1.7rem;
}
