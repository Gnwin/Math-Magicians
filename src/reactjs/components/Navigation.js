import React, { useState } from 'react';
import { Link } from 'react-router-dom';

const Navigation = () => {
  const [navContent] = useState({
    title: 'Math magicians',
    links: ['Home', 'Calculator', 'Quote']
  })

  return (
    <div>
      <div>{navContent.title}</div>
      <div>
        {navContent.links.map((link, index) => {(
          <Link to={`./${link.toLowerCase()}`} key={index}>{link}</Link>
        )})}
      </div>
    </div>
  )
}

export default Navigation