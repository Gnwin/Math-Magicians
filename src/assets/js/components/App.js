import React from 'react';
import '../../stylesheets/App.css';
import Calculator from './Calculator';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="app">
        <Calculator />
      </div>
    );
  }
}

export default App;
