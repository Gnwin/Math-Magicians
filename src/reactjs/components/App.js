import React, { useState, useEffect } from 'react';
import '../../assets/stylesheets/App.css';
import Calculator from './Calculator';
import numbers from './numbers';
import operands from './operands';

const App = () => {
  const [calcNumbers, setNumbers] = useState([]);
  const [calcOperands, setOperands] = useState([]);

  useEffect(() => {
    setNumbers(numbers);
    setOperands(operands);
  }, []);

  return (
    <div className="app">
      <Calculator operands={calcOperands} numbers={calcNumbers} />
    </div>
  );
};

export default App;
