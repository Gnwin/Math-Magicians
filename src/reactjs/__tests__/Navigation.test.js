import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import Navigation from '../components/Navigation';
import App from '../components/App';

describe('it tests the Navigation Component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<BrowserRouter><Navigation /></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('checks if the Navigation Component is in the document', () => {
    render(<BrowserRouter><Navigation /></BrowserRouter>);
    const logo = screen.getByText('Math magicians');
    expect(logo).toBeInTheDocument();
  });

  it('renders a navigation Component', () => {
    render(<BrowserRouter><Navigation /></BrowserRouter>);
    const navElement = screen.getByTestId('nav-a');
    expect(navElement).toHaveClass('nav');
  });

  it('checks if there a Home link in the navigation component', () => {
    render(<BrowserRouter><Navigation /></BrowserRouter>);
    const homeLink = screen.getByText('Home');
    expect(homeLink.closest('a')).toHaveAttribute("href", "/home")
  });

  it('checks if there a Calculator link in the navigation component', () => {
    render(<BrowserRouter><Navigation /></BrowserRouter>);
    const calculatorLink = screen.getByText('Calculator');
    expect(calculatorLink.closest('a')).toHaveAttribute("href", "/calculator")
  });

  it('checks if there a Quote link in the navigation component', () => {
    render(<BrowserRouter><Navigation /></BrowserRouter>);
    const quoteLink = screen.getByText('Quote');
    expect(quoteLink.closest('a')).toHaveAttribute("href", "/quote")
  });

  it('renders the home page when the user clicks the home link', () => {
    render(<BrowserRouter><App /></BrowserRouter>);
    const homeLink = screen.getByText('Home');
    fireEvent.click(homeLink);
    const homeContainer = screen.getByTestId('home-1');
    const firstchild = homeContainer.childNodes[0];
    expect(firstchild).toHaveClass('home-title');
  });

  it('renders the calculator page when the user clicks the calculator link', () => {
    render(<BrowserRouter><App /></BrowserRouter>);
    const calculatorLink = screen.getByText('Calculator');
    fireEvent.click(calculatorLink);
    const calculatorContainer = screen.getByTestId('calc');
    expect(calculatorContainer).toHaveClass('calculator');
  });

  it('renders the Quote page when the user clicks the quote link', () => {
    render(<BrowserRouter><App /></BrowserRouter>);
    const quoteLink = screen.getByText('Quote');
    fireEvent.click(quoteLink);
    const quoteContainer = screen.getByTestId('quote-1');
    expect(quoteContainer).toHaveClass('quote');
  });
});
