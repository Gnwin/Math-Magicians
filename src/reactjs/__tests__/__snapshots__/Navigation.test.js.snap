// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it tests the Navigation Component renders correctly 1`] = `
<div
  className="nav"
  data-testid="nav-a"
>
  <div
    className="nav-title"
  >
    Math magicians
  </div>
  <div
    className="nav-links"
    data-testid="nav-b"
  >
    <a
      className="nav-link"
      data-testid="nav-1"
      href="/home"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="nav-link"
      data-testid="nav-2"
      href="/calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
    <a
      className="nav-link"
      data-testid="nav-3"
      href="/quote"
      onClick={[Function]}
    >
      Quote
    </a>
  </div>
</div>
`;
